python/lsst/dax/imgserv/MetadataFitsDb.py

```diff
@@ -294,73 +284,58 @@ def insertMetadataFits(self, metadata):
         try:
-            sql = ("SELECT 1 FROM FitsFiles WHERE "
-                   "fileName = %s")
-            self._log.debug(sql, str(fileName))
-            cursor.execute(sql, fileName)
-            r = cursor.fetchall()
-            # Nothing found, so it needs to be added.
-            if len(r) < 1:
-                #If this fails for any reason, we do not want the database altered.
-                cursor.execute("START TRANSACTION")
-                cursor.execute("SET autocommit = 0")
-                # Insert the file into the file table.
-                colVal = [("fileName", fileName), ("hduCount", hdus)]
-                executeInsertList(cursor, "FitsFiles", colVal, self._log)
-                lastFitsFileId = cursor.lastrowid
-                for key, entry in entries.iteritems():
-                    value   = entry[0]
-                    lineNum = entry[1]
-                    comment = entry[2]
-                    # Put in one entry for each element of the tuple
-                    if isinstance(value, tuple):
-                        num = lineNum
-                        for v in value:
-                            # scanFile should be skipping line numbers when it sees a tuple,
-                            # so that there are no duplicates.
-                            self._insertFitsValue(cursor, lastFitsFileId, key, v, num, comment)
-                            num += 1
-                    else:
-                        self._insertFitsValue(cursor, lastFitsFileId, key, value, lineNum, comment)
-                for hdu in range(1,hdus+1):
-                    metadataPosition = MetadataPosition(lastFitsFileId, hdu, cursor, entries)
-                    metadataPosition._insert()
-                cursor.execute("COMMIT")
-        except MySQLdb.Error as err:
-            cursor.execute("ROLLBACK")
-            self._log.warn( "ROLLBACK due to ERROR MySQLdb %s --%s", err, sql)
-            quit() # TODO delete this line, for now it is good to stop and examine these
-        cursor.close()
+            with self._conn.begin() as trans:
+                sql = ("SELECT 1 FROM FitsFiles WHERE "
+                       "fileName = %s")
+                self._log.debug(sql, str(fileName))
+                results = self._conn.execute(sql, fileName)
+                r = results.fetchall()
+                # Nothing found, so it needs to be added.
+                if len(r) < 1:
+                    #If this fails for any reason, we do not want the database altered.
+                    # Insert the file into the file table.
+                    colVal = [("fileName", fileName), ("hduCount", hdus)]
+                    lastFitsFileId = executeInsertList(self._conn, "FitsFiles", colVal, self._log)
+                    for key, entry in entries.iteritems():
+                        value   = entry[0]
+                        lineNum = entry[1]
+                        comment = entry[2]
```

Comment:
I think more Pythonic is `value, lineNum, comment = entry` (or `entry[:3]` if entry can be longer)